src/main/java/bio/terra/workspace/app/configuration/ApiResourceConfig.java

```diff
@@ -28,7 +28,12 @@ public void addInterceptors(InterceptorRegistry registry) {
           public boolean preHandle(
               HttpServletRequest request, HttpServletResponse response, Object handler)
               throws Exception {
-            spanCustomizer.tag("session-id", request.getSession().getId());
+            String mdcRequestId = request.getHeader("X-Request-ID");
+            if (mdcRequestId != null) spanCustomizer.tag("mdc-request-id", mdcRequestId);
+
+            String mdcCorrelationId = request.getHeader("X-Correlation-ID");
+            if (mdcCorrelationId != null)
+              spanCustomizer.tag("mdc-correlation-id", mdcCorrelationId);
```

Comment:
hm, is there a way to reuse the implementation in `MdcLogEnhancerFilter`? That filter also generates a unique MDC id if none exists, it would be good to have the generated id if neither header is present.

Reply:
I believe `addinterceptor` is being called before `MdcLogEnhancerFilter`. I think the only way to get id found/generated in the filter to be available in the interceptor is to move the MDC log code from the filter to the interceptor. 

That way, we get the MDC id here, create it if it doesn't exist and then add the tag to the customizer, all in the interceptor. 

Unsure if there's a better way to do it than that.